src/http/request.cpp

```diff
@@ -53,7 +66,8 @@ void Request::serialize(net::Buffer &buff) {
         buff << kv.first << ": " << kv.second << "\r\n";
     }
     buff << "Host: " << url.address;
-    if (url.port != 80) {
+    if ((url.schema == "http" and url.port != 80) or
+         (url.schema == "https" and url.port != 443)) {
```

Comment:
@hellais one issue with fsrn.org was that we constructed the URL contained `443` and this lead to a redirect loop; the fix was to make sure no port was added when the port is the default one for a schema.